modules/javafx.graphics/src/main/java/javafx/css/StyleableBooleanProperty.java

```diff
@@ -70,7 +70,8 @@ public StyleableBooleanProperty(boolean initialValue) {
     /** {@inheritDoc} */
     @Override
     public void applyStyle(StyleOrigin origin, Boolean v) {
-        TransitionDefinition transition = this.origin != null && getBean() instanceof Node node ?
+        // If the value is applied for the first time, we don't start a transition.
+        TransitionDefinition transition = getBean() instanceof Node node && !NodeHelper.isInitialCssState(node) ?
```

Comment:
minor: to simplify the code in many places, you could change `isInitialState` to accept an `Object` argument instead of `Node` and do an `instanceof` there.

Reply:
I don't think this would work, since we need the `Node`-typed `node` variable as an argument for `NodeHelper.findTransitionDefinition`, so we can't remove the `instanceof` test here.

Reply:
NodeHelper:

```
    public static boolean isInitialCssState(Object node) {
        if(node instanceof Node n) {
            return nodeAccessor.isInitialCssState(n);
        }
        return false;
    }
```

Reply:
or better yet, move the instanceof to NodeHelper.findTransitionDefinition()

Reply:
I'll see whether I can refactor these methods a bit, but not as part of this PR.